conv.go

```diff
@@ -0,0 +1,196 @@
+package DateTimeMate
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const (
+	secondsPerNanoseconds = 0.000000001
+	secondsPerMicrosecond = 0.000001
+	secondsPerMillisecond = 0.001
+	secondsPerMinute      = 60
+	secondsPerHour        = 60 * secondsPerMinute
+	secondsPerDay         = 24 * secondsPerHour
+	secondsPerWeek        = 7 * secondsPerDay
+	secondsPerYear        = 365.25 * secondsPerDay
+)
+
+var unitMap = map[string]float64{
+	"nanosecond":  secondsPerNanoseconds,
+	"microsecond": secondsPerMicrosecond,
+	"millisecond": secondsPerMillisecond,
+	"second":      1,
+	"minute":      secondsPerMinute,
+	"hour":        secondsPerHour,
+	"day":         secondsPerDay,
+	"week":        secondsPerWeek,
+	"year":        secondsPerYear,
+}
+
+var unitBriefMap = map[string]string{
+	"ns": "nanosecond",
+	"us": "microsecond",
+	"ms": "millisecond",
+	"s":  "second",
+	"m":  "minute",
+	"h":  "hour",
+	"D":  "day",
+	"W":  "week",
+	"Y":  "year",
+}
+
+type Conv struct {
+	Source string
+	Target string
+	Brief  bool
+}
+
+type OptionsConv func(*Conv)
+
+func NewConv(options ...OptionsConv) *Conv {
+	conv := &Conv{}
+	for _, opt := range options {
+		opt(conv)
+	}
+	return conv
+}
+
+func ConvWithSource(source string) OptionsConv {
+	return func(conv *Conv) {
+		conv.Source = source
+	}
+}
+
+func ConvWithTarget(target string) OptionsConv {
+	return func(conv *Conv) {
+		conv.Target = target
+	}
+}
+
+func ConvWithBrief(brief bool) OptionsConv {
+	return func(conv *Conv) {
+		conv.Brief = brief
+	}
+}
+
+func (conv *Conv) parseSource() (float64, error) {
+	parts := strings.Fields(conv.Source)
+	totalSeconds := 0.0
+
+	for i := 0; i < len(parts); i += 2 {
+		value, err := strconv.ParseFloat(parts[i], 64)
+		if err != nil {
+			return 0, err
+		}
+		unit := strings.ToLower(strings.TrimSuffix(parts[i+1], "s")) // Remove plural 's'
+		if seconds, ok := unitMap[unit]; ok {
+			totalSeconds += value * seconds
+		}
+	}
+	return totalSeconds, nil
+}
+
+func (conv *Conv) formatTarget(seconds float64, units []string) string {
+	result := ""
+	for _, unit := range units {
+		unitInSeconds := unitMap[strings.TrimSuffix(unit, "s")]
+		value := seconds / unitInSeconds
+		intValue := int(value)
+		if intValue > 0 {
+			if intValue == 1 {
+				unit = removeTrailingS(unit)
+			} else if unit[len(unit)-1] != 's' {
+				unit += "s"
+			}
+			result += fmt.Sprintf("%d %s ", intValue, unit)
+			seconds -= float64(intValue) * unitInSeconds
+		}
```

Comment:
```suggestion
      if intValue <= 0 {
            continue
      }

      unit = removeTrailingS(unit)
      if intValue > 1 {
            unit += "s"
      }
      result += fmt.Sprintf("%d %s ", intValue, unit)
      seconds -= float64(intValue) * unitInSeconds
```

Reply:
This is much easier to read & follow - added. Thanks.